common/persistence/data_store_interfaces.go

```diff
@@ -905,6 +905,14 @@ type (
 	}
 )
 
+func (tr *InternalGetHistoryTreeResponse) ByBranchID() map[string]*types.HistoryBranch {
+	out := make(map[string]*types.HistoryBranch)
```

Comment:
```suggestion
	out := make(map[string]*types.HistoryBranch, len(tr.Branches))
```
might as well pre-size when it's easy